hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientMapProxy.java

```diff
@@ -2191,4 +2193,30 @@ private V mergeLocally(K key, V value, BiFunction<? super V, ? super V, ? extend
         }
     }
 
+    @Override
+    public void replaceAll(@Nonnull BiFunction<? super K, ? super V, ? extends V> function) {
+        checkNotNull(function, NULL_BIFUNCTION_IS_NOT_ALLOWED);
+        replaceAllInternal(function);
+    }
+
+    protected void replaceAllInternal(BiFunction<? super K, ? super V, ? extends V> function) {
+        if (SerializationUtil.isClassStaticAndSerializable(function)) {
+            int partitionCount = getContext().getPartitionService().getPartitionCount();
+            List<Future<ClientMessage>> futures = new ArrayList<>(partitionCount);
+            Data functionAsData = toData(function);
+            for (int partitionId = 0; partitionId < partitionCount; partitionId++) {
+                ClientMessage request = MapReplaceAllCodec.encodeRequest(name, functionAsData);
+                futures.add(new ClientInvocation(getClient(), request, getName(), partitionId).invoke());
```

Comment:
It could be the case that client version is 5.1 or above, the server version is 5.0 or above. 
In such cases, the server throws `UnsupportedOperationException` since  `MapReplaceAllMessageTask` does not exist on the servers yet. 
We want our clients to work in this cases. Can you catch this exception and fall back to `IMap.super.replaceAll(function);` implementation.
 And secondly, we don't want to get this expcetion always. After realizing that server can not work with the new implementation, we can cache this value on the proxy and use always fallback after that point. 
 
 To test this you can use HazelcastStarter test utility where you can create old members on the same jvm. See 
https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/test/java/com/hazelcast/client/starter/HazelcastClientStarterTest.java#L46
It is beter to make this test seperate than ClientMapBasicTest because it will be a slow one. You can create a new test and use the same annotations that we used on  `HazelcastClientStarterTest` to denote it is slow. 
We are separating those tests from the pr-builder and running them nightly not on every pr. That is why we have these annotations.

Reply:
Done